### PR TITLE
Changed API logging feature on CLI

### PR DESCRIPTION
**Pull request by amulyavarote**
Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

# Description

Update api logging feature - Changed from api-log-level to enable-api-logging.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: NA

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -588,14 +588,13 @@ $ dapr invoke --app-id nodeapp --unix-domain-socket --method mymethod
 
 ### Set API log level
 
-In order to set the Dapr runtime API calls log verbosity level, use the `api-log-level` flag:
+In order to set the Dapr runtime API calls log verbosity, use the `enable-api-logging` flag:
```
Comment (mukundansundar) [inline README.md:591]:
```suggestion
In order to set the Dapr runtime to log API calls with `INFO` log verbosity, use the `enable-api-logging` flag:
```

File: README.md [hunk 2]
```diff
@@ -588,14 +588,13 @@ $ dapr invoke --app-id nodeapp --unix-domain-socket --method mymethod
...
 
 ```bash
-dapr run --app-id nodeapp --app-port 3000 node app.js --api-log-level info
+dapr run --app-id nodeapp --app-port 3000 node app.js enable-api-logging true
```
Comment (mukundansundar) [inline README.md:594]:
```suggestion
dapr run --app-id nodeapp --app-port 3000 node app.js enable-api-logging
```
true actually should not be needed, check examples similar to `enable-mtls`

File: cmd/run.go [hunk 3]
```diff
@@ -365,7 +365,7 @@ func init() {
 	RunCmd.Flags().BoolP("help", "h", false, "Print this help message")
 	RunCmd.Flags().IntVarP(&maxRequestBodySize, "dapr-http-max-request-size", "", -1, "Max size of request body in MB")
 	RunCmd.Flags().StringVarP(&unixDomainSocket, "unix-domain-socket", "u", "", "Path to a unix domain socket dir. If specified, Dapr API servers will use Unix Domain Sockets")
-	RunCmd.Flags().StringVarP(&apiLogLevel, "api-log-level", "", "debug", "The api calls log verbosity. Valid values are: debug, info, off")
+	RunCmd.Flags().BoolVar(&enableApiLogging, "enable-api-logging", false, "The api calls log verbosity. Valid values are: true or false")
```
Comment (mukundansundar) [inline cmd/run.go:368]:
```suggestion
	RunCmd.Flags().BoolVar(&enableApiLogging, "enable-api-logging", false, "Log API calls at INFO verbosity. Valid values are: true or false")
```

File: tests/e2e/standalone/standalone_test.go [hunk 4]
```diff
@@ -456,17 +456,17 @@ func testRun(t *testing.T) {
...
 
-func testRunApiLogLevel(t *testing.T) {
+func testRunEnableApiLogging(t *testing.T) {
 	daprPath := getDaprPath()
 	args := []string{
 		"run",
-		"--app-id", "apiloglevel_info",
-		"--api-log-level", "info",
+		"--app-id", "enableApiLogging_info",
+		"--enable-api-logging", true,
```
Comment (mukundansundar) [inline tests/e2e/standalone/standalone_test.go:464]: no need to set `true` ideally...

File: tests/e2e/standalone/standalone_test.go [hunk 5]
```diff
@@ -477,17 +477,16 @@ func testRunApiLogLevel(t *testing.T) {
 
 	args = []string{
 		"run",
-		"--app-id", "apiloglevel_debug",
-		"--api-log-level", "debug",
+		"--app-id", "enableApiLogging_info",
+		"--enable-api-logging", false,
```
Comment (mukundansundar) [inline tests/e2e/standalone/standalone_test.go:481]: by default false ... so no need to mention that here ...

File: tests/e2e/standalone/standalone_test.go [hunk 6]
```diff
@@ -477,17 +477,16 @@ func testRunApiLogLevel(t *testing.T) {
...
 		"--log-level", "debug",
```
Comment (mukundansundar) [inline tests/e2e/standalone/standalone_test.go:482]: no need debug log level...

File: tests/e2e/standalone/standalone_test.go [hunk 7]
```diff
@@ -477,17 +477,16 @@ func testRunApiLogLevel(t *testing.T) {
...
 		"--", "bash", "-c", "echo 'test'",
 	}
 
-	t.Run(fmt.Sprintf("check apiloglevel flag debug mode"), func(t *testing.T) {
+	t.Run(fmt.Sprintf("check enableApiLogging flag in disabled mode"), func(t *testing.T) {
 		output, err := spawn.Command(daprPath, args...)
 		t.Log(output)
 		require.NoError(t, err, "run failed")
-		assert.Contains(t, output, "level=debug msg=\"HTTP API Called: PUT /v1.0/metadata/appCommand\"")
```
Comment (mukundansundar) [inline tests/e2e/standalone/standalone_test.go:490]: do a negative assertion here

File: tests/e2e/standalone/standalone_test.go [hunk 8]
```diff
@@ -477,17 +477,14 @@ func testRunApiLogLevel(t *testing.T) {
...
+		"--app-id", "enableApiLogging_info",
 		"--", "bash", "-c", "echo 'test'",
 	}
 
-	t.Run(fmt.Sprintf("check apiloglevel flag debug mode"), func(t *testing.T) {
+	t.Run(fmt.Sprintf("check enableApiLogging flag in disabled mode"), func(t *testing.T) {
 		output, err := spawn.Command(daprPath, args...)
 		t.Log(output)
 		require.NoError(t, err, "run failed")
-		assert.Contains(t, output, "level=debug msg=\"HTTP API Called: PUT /v1.0/metadata/appCommand\"")
```
Comment (mukundansundar) [inline tests/e2e/standalone/standalone_test.go:490]: do not remove this assert.
Make sure that line is not there in the output. 
Do a negative assertion.

**Review by mukundansundar — CHANGES_REQUESTED** [review]

**Review by mukundansundar — CHANGES_REQUESTED** [review]
@amulyavarote change to release-1.7 branch ....

**Review by pravinpushkar — APPROVED** [review]
LGTM. 
Docs update PR link can be linked.

**Review by artursouza — APPROVED** [review]

**Review by mukundansundar — APPROVED** [review]
lgtm

**Pull request merged**